From c781b2884c9dde0aea49549caf1831123bb8fe8b Mon Sep 17 00:00:00 2001
From: Arthur Liberman <arthur_liberman@hotmail.com>
Date: Sat, 15 Mar 2025 01:53:10 +0200
Subject: [PATCH 09/17] Revert "alsa: drop obsolete
 'aml_set_audio_passthrough'"

This reverts commit eb97f5c837e174b60b86c3d74405ce49400fa2f2.
---
 xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp | 1 +
 xbmc/utils/AMLUtils.cpp                     | 5 +++++
 xbmc/utils/AMLUtils.h                       | 1 +
 3 files changed, 7 insertions(+)

diff --git a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
index 3b0453bfce..e137df228e 100644
--- a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
+++ b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
@@ -807,6 +807,7 @@ bool CAESinkALSA::Initialize(AEAudioFormat &format, std::string &device)
       }
     }
 
+    aml_set_audio_passthrough(m_passthrough);
     aml_configure_simple_control(device, codec);
   }
 
diff --git a/xbmc/utils/AMLUtils.cpp b/xbmc/utils/AMLUtils.cpp
index f4beec321a..697ba6b7c6 100644
--- a/xbmc/utils/AMLUtils.cpp
+++ b/xbmc/utils/AMLUtils.cpp
@@ -275,6 +275,11 @@ bool aml_convert_to_dv_by_vs_engine(StreamHdrType hdrType)
   return ((convert_to_dv && !!user_convert_to_dv && !!dv_user_enabled) == 1);
 }
 
+void aml_set_audio_passthrough(bool passthrough)
+{
+  CSysfsPath("/sys/class/audiodsp/digital_raw", (passthrough ? 2 : 0));
+}
+
 void aml_set_3d_video_mode(unsigned int mode, bool framepacking_support, int view_mode)
 {
   int fd;
diff --git a/xbmc/utils/AMLUtils.h b/xbmc/utils/AMLUtils.h
index 91534d9402..d369c70f7a 100644
--- a/xbmc/utils/AMLUtils.h
+++ b/xbmc/utils/AMLUtils.h
@@ -80,6 +80,7 @@ bool aml_support_dolby_vision();
 bool aml_dolby_vision_enabled();
 bool aml_convert_to_dv_by_vs_engine(StreamHdrType hdrType);
 bool aml_has_frac_rate_policy();
+void aml_set_audio_passthrough(bool passthrough);
 void aml_set_3d_video_mode(unsigned int mode, bool framepacking_support, int view_mode);
 bool aml_mode_to_resolution(const char *mode, RESOLUTION_INFO *res);
 bool aml_get_native_resolution(RESOLUTION_INFO *res);
-- 
2.48.1

